cmd/kube-proxy/app/server.go

```diff
@@ -146,6 +152,13 @@ func (s *ProxyServer) Run(_ []string) error {
 		}, 5*time.Second)
 	}
 
+	go func() {
+		err := proxy.ServeDNS(dnsHandler)
```

Comment:
Ok - add a comment to that effect here "cannot restart skydns without
leaking goroutines"

On Jul 21, 2015, at 2:08 PM, Solly <notifications@github.com> wrote:

In cmd/kube-proxy/app/server.go
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11599#discussion_r35134201>
:

> @@ -146,6 +152,13 @@ func (s *ProxyServer) Run(_ []string) error {
>  		}, 5*time.Second)
>  	}
>
> +	go func() {
> +		err := proxy.ServeDNS(dnsHandler)

SkyDNS spawns goroutines in Run() --
https://github.com/skynetservices/skydns/blob/master/server/server.go#L149

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11599/files#r35134201>
.